{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dans le dossier d'analyse, faire un dossier queries et y placer les sequences proteiques des genes\n",
    "# de virulence que l'on souhaite tester.\n",
    "# NB. Les sequences peuvent etre des fichiers multi-fasta (plusieurs versions d'un meme gene)\n",
    "# ou encore des fichiers fasta avec une seule séquence\n",
    "# Par exemple, le fichier eae.fasta.fasta contient différentes versions de gènes d'intimins (beta, epsilon, etc)\n",
    "# bfp.fasta contient les genes bfp tires d'un plasmide\n",
    "# ehxA contient la sequence de reference pour l'hemolysine\n",
    "# nommer correctement ces genes, car ils seront utilises dans la sortie du programme"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import glob\n",
    "import re\n",
    "from pathlib import Path\n",
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastpCommandline\n",
    "#from Bio.Alphabet import IUPAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_files(start_dir, file_type):\n",
    "    \"\"\" Given a directory, make a list of files\n",
    "    This function will find all files recursively.\n",
    "    \"\"\"\n",
    "    files = []\n",
    "    for filename in Path(start_dir).glob('**/*.' + file_type):\n",
    "        files.append(str(filename))\n",
    "    return files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Preparation of a database of CDS from all strains!\n",
    "cwd = os.getcwd()\n",
    "if not os.path.exists(cwd + '/database'):\n",
    "    os.mkdir(cwd + '/database')\n",
    "if not os.path.exists(cwd + '/xml'):\n",
    "    os.mkdir(cwd + '/xml')\n",
    "    \n",
    "gbk_files = get_files(\"./MS_minION_all_gbk\", \"gbk\")\n",
    "\n",
    "# Doing it on a bunch of gbk files\n",
    "for gbk_i, gbk_file in enumerate(gbk_files):\n",
    "    \n",
    "    # Work on UNIX filesystems, name the fasta file from the gbk file\n",
    "    outfile_name = os.path.basename(gbk_file).replace('.gbk', '')\n",
    "    print('Adding ', outfile_name, sep=' ')\n",
    "    \n",
    "    # Prepare a fasta file for writing\n",
    "    with open(\"database/\" + outfile_name + '.fasta', \"w\") as output_handle:\n",
    "        records = SeqIO.parse(gbk_file, \"genbank\")\n",
    "\n",
    "        for record in records:\n",
    "            for i_feature, feature in enumerate(record.features):\n",
    "            \n",
    "                if feature.type == 'CDS':\n",
    "                    if 'gene' in feature.qualifiers: # if a gene tag is present use it the name\n",
    "                        feature_id = feature.qualifiers['gene'][0] + '_' + record.name +  '_' + outfile_name\n",
    "                    else:\n",
    "                        feature_id = feature.qualifiers['locus_tag'][0] + '_' + record.name + '_' + outfile_name\n",
    "                \n",
    "                    # Create the sequence object\n",
    "                    my_seq = Seq(feature.qualifiers['translation'][0],IUPAC.protein)\n",
    "                \n",
    "                    # Create a SeqRecord object with the protein sequence\n",
    "                    simple_seq_r = SeqRecord(my_seq, id=feature_id, description = '')\n",
    "                    SeqIO.write(simple_seq_r, output_handle, \"fasta\")\n",
    "                    \n",
    "                    >>> simple_seq_r = SeqRecord(simple_seq, id=\"AC12345\")\n",
    "                    \n",
    "                    \n",
    "                    \n",
    "                    \n",
    "                    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### In a terminal prepare a NCBI blast+ db\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "### cat *.fasta > MS_strains_v2\n",
    "### makeblastdb -in t6ss.fasta -input_type fasta  -dbtype prot -title T6SS_stuff -out T6SS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Perform blastp searches against the MS_strains database\n",
    "# The MS_strains NCBI+ database contains protein sequences from the genomes of the 244 strains included in the manuscript \n",
    "# Output will be in XML folders\n",
    "# Create xml folder if it not exists\n",
    "cwd = os.getcwd()\n",
    "if not os.path.exists(cwd + '/xml'):\n",
    "    os.mkdir(cwd + '/xml')\n",
    "\n",
    "# First grab the query files in the queries folder\n",
    "virulence_factors = get_files (\"./queries/integrase_detection\", \"fasta\")\n",
    "\n",
    "# Next lauch blastp searches from biopython\n",
    "for f in virulence_factors:\n",
    "    outname =  os.path.basename(f).replace('.fasta','')\n",
    "    blastp_cline = NcbiblastpCommandline(query = f, db=\"MS_strains_v2\", evalue=1e-6, outfmt=5, out='xml/' + outname + '.xml')\n",
    "    stdout, stderr = blastp_cline()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing int_yicC\n",
      "* Query #1 is Int_yicC_PmiNLAE_NZ_FNAA01000003*(length=403) \n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|945845 intZ_3_chr_tig63_Res13-Lact-PEA02-11_chromosome\n",
      "length: 419\n",
      "e value: 7.15513e-94\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDS-GSPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    K+DG GLY   P +  +P+W  RY + GK++ M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKAGDRFDGKADGNGLYICYPKNYTAPFWRFRYKLAGKQRAMVIGSYSELSLSKARETAKELSARV...\n",
      "kk421\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|158857 intA_chr_tig35_R13-AF31-pSow-05_chromosome\n",
      "length: 421\n",
      "e value: 1.29002e-93\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAGE    ++DG GLY   P+  S P+W  RY + GK++ M +G Y+D+SL+ AR   +     +...\n",
      "LTDIQIRAWIKAGEHFEGRADGNGLYLCFPEKYSAPFWRYRYRMAGKQRTMHIGSYSDLSLSKARATVKELAARV...\n",
      "kk421\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|144055 intA_1_chr_tig1273_R13-AF31-pSow-02_chromosome\n",
      "length: 421\n",
      "e value: 1.29002e-93\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAGE    ++DG GLY   P+  S P+W  RY + GK++ M +G Y+D+SL+ AR   +     +...\n",
      "LTDIQIRAWIKAGEHFEGRADGNGLYLCFPEKYSAPFWRYRYRMAGKQRTMHIGSYSDLSLSKARATVKELAARV...\n",
      "kk421\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|132621 intA_chr_tig33_R13-AF22-pSow-05_chromosome\n",
      "length: 421\n",
      "e value: 1.29002e-93\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAGE    ++DG GLY   P+  S P+W  RY + GK++ M +G Y+D+SL+ AR   +     +...\n",
      "LTDIQIRAWIKAGEHFEGRADGNGLYLCFPEKYSAPFWRYRYRMAGKQRTMHIGSYSDLSLSKARATVKELAARV...\n",
      "kk421\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|127689 intA_chr_tig34_R13-AF22-pSow-04_chromosome\n",
      "length: 421\n",
      "e value: 1.29002e-93\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAGE    ++DG GLY   P+  S P+W  RY + GK++ M +G Y+D+SL+ AR   +     +...\n",
      "LTDIQIRAWIKAGEHFEGRADGNGLYLCFPEKYSAPFWRYRYRMAGKQRTMHIGSYSDLSLSKARATVKELAARV...\n",
      "kk421\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|122691 intA_chr_tig36_R13-AF22-pSow-03_chromosome\n",
      "length: 421\n",
      "e value: 1.29002e-93\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAGE    ++DG GLY   P+  S P+W  RY + GK++ M +G Y+D+SL+ AR   +     +...\n",
      "LTDIQIRAWIKAGEHFEGRADGNGLYLCFPEKYSAPFWRYRYRMAGKQRTMHIGSYSDLSLSKARATVKELAARV...\n",
      "kk439\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|238093 intZ_1_chr_tig27_R13-CV1-pSow-02_chromosome\n",
      "length: 439\n",
      "e value: 2.69432e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk422\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|936869 intA_5_chr_tig105_Res13-Lact-LER2-35-b_chromosome\n",
      "length: 422\n",
      "e value: 2.8647e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAGE    ++DG GLY    ++  +P W  RY + GK + M +G YA++SLA AR  A+     +...\n",
      "LTDMQIRAWIKAGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYAELSLAKARETAKELSARV...\n",
      "kk439\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1355021 intZ_2_chr_tig117_Res13-Sevr-PEB15-13_chromosome\n",
      "length: 439\n",
      "e value: 3.92932e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk439\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1128272 intZ_4_chr_tig69_Res13-Lact-PEB18-16_chromosome\n",
      "length: 439\n",
      "e value: 3.92932e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk439\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1115845 intZ_1_chr_tig4_Res13-Lact-PEB17-18_chromosome\n",
      "length: 439\n",
      "e value: 3.92932e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk439\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|909139 intZ_1_chr_tig3_Res13-Lact-LEB3-27_chromosome\n",
      "length: 439\n",
      "e value: 3.92932e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk439\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|905698 intZ_1_chr_tig33_Res13-Lact-LEB3-20_chromosome\n",
      "length: 439\n",
      "e value: 3.92932e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk439\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|899493 intZ_1_chr_tig2_Res13-Lact-LEB3-12_chromosome\n",
      "length: 439\n",
      "e value: 3.92932e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk439\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|893730 intZ_4_chr_tig68_Res13-Lact-LEB3-02_chromosome\n",
      "length: 439\n",
      "e value: 3.92932e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk439\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|465119 intZ_1_chr_tig4_Res13-Croi-PEB17-01_chromosome\n",
      "length: 439\n",
      "e value: 3.92932e-90\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1007263 intZ_3_chr_tig60_Res13-Lact-PEA13-26_chromosome\n",
      "length: 419\n",
      "e value: 1.47302e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1002739 intZ_3_chr_tig65_Res13-Lact-PEA13-25_chromosome\n",
      "length: 419\n",
      "e value: 1.47302e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|998012 intZ_3_chr_tig66_Res13-Lact-PEA13-17_chromosome\n",
      "length: 419\n",
      "e value: 1.47302e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|993471 intZ_3_chr_tig65_Res13-Lact-PEA13-11_chromosome\n",
      "length: 419\n",
      "e value: 1.47302e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|988930 intZ_3_chr_tig63_Res13-Lact-PEA13-10_chromosome\n",
      "length: 419\n",
      "e value: 1.47302e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|984313 intZ_3_chr_tig62_Res13-Lact-PEA13-09_chromosome\n",
      "length: 419\n",
      "e value: 1.47302e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk420\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1380952 intA_2_chr_tig14_Res13-Sevr-PEC09-47_chromosome\n",
      "length: 420\n",
      "e value: 2.74612e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDS-GSPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "MT  Q+ A  +A E    +SDG GL    P     P+W  RY   GK++ M +G YA++SLA AR  A+     +...\n",
      "MTDIQIKAWIRADERFEARSDGNGLSLCFPKKYAVPFWRFRYRFSGKQRVMGIGSYAELSLAKARETAKELSARV...\n",
      "kk420\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1377143 intA_2_chr_tig15_Res13-Sevr-PEC09-46_chromosome\n",
      "length: 420\n",
      "e value: 2.74612e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDS-GSPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "MT  Q+ A  +A E    +SDG GL    P     P+W  RY   GK++ M +G YA++SLA AR  A+     +...\n",
      "MTDIQIKAWIRADERFEARSDGNGLSLCFPKKYAVPFWRFRYRFSGKQRVMGIGSYAELSLAKARETAKELSARV...\n",
      "kk420\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1372803 intA_2_chr_tig18_Res13-Sevr-PEC09-36_chromosome\n",
      "length: 420\n",
      "e value: 2.74612e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDS-GSPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "MT  Q+ A  +A E    +SDG GL    P     P+W  RY   GK++ M +G YA++SLA AR  A+     +...\n",
      "MTDIQIKAWIRADERFEARSDGNGLSLCFPKKYAVPFWRFRYRFSGKQRVMGIGSYAELSLAKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|278520 intA_2_chr_tig139_R13-CV2-pWea-05_chromosome\n",
      "length: 419\n",
      "e value: 5.55186e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|14442 intA_1_chr_tig50_R13-AF11-pFin-04_chromosome\n",
      "length: 419\n",
      "e value: 6.11783e-89\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|83433 intA_chr_tig61_R13-AF21-pWea-04_chromosome\n",
      "length: 419\n",
      "e value: 1.61491e-88\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1073559 intA_3_chr_tig166_Res13-Lact-PEB07-13-B_chromosome\n",
      "length: 419\n",
      "e value: 5.06478e-88\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSGS-PYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAG+    ++DG GLY   P + + P+W  RY   GK++ M +G YA++SL+ AR  A+     +...\n",
      "ITDMQIRAWIKAGDRFEGRADGNGLYICYPKNYTVPFWRFRYKFAGKQRAMVIGSYAELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|249909 intZ_3_chr_tig87_R13-CV1-pSow-04_chromosome\n",
      "length: 419\n",
      "e value: 4.82024e-87\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLY--FVVPDSGSPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKRE...\n",
      "+T  Q+ A  KAG     KSDG GLY  +   D+ SP W  RY   GK + MT+G Y  +SLA AR         ...\n",
      "LTDIQIKAWIKAGTRFEGKSDGNGLYLRYRTTDT-SPSWRFRYRFAGKPRVMTIGAYGVISLAKAREITRELAAR...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1177362 intZ_chr_tig61_Res13-Lact-PEC11-32_chromosome\n",
      "length: 419\n",
      "e value: 6.1101e-87\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  KAGE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKAGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk437\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|624575 intA_chr_tig146_Res13-Fini-PEC06-39_chromosome\n",
      "length: 437\n",
      "e value: 1.42308e-86\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLYFVVPDSG-SPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKREL...\n",
      "+T  Q+ A  K+GE    ++DG GLY    ++  +P W  RY + GK + M +G Y+++SL+ AR  A+     +...\n",
      "LTDMQIRAWIKSGERFEGRADGNGLYLRYREADKTPTWRFRYKLAGKSRAMLIGSYSELSLSKARETAKELSARV...\n",
      "kk419\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|49131 intZ_chr_tig108_R13-AF12-pSow-01_chromosome\n",
      "length: 419\n",
      "e value: 2.48019e-86\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLY--FVVPDSGSPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKRE...\n",
      "+T  Q+ A  KAG     KSDG GLY  +   D+ SP W  RY   GK + MT+G Y  +SLA AR         ...\n",
      "LTDIQIKAWIKAGTRFEGKSDGNGLYLRYRTTDT-SPSWRFRYRFAGKPRVMTIGAYGVISLAKAREITRELAAR...\n",
      "kk418\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1400741 intZ_2_chr_tig86_Res13-Sevr-PER08-38_chromosome\n",
      "length: 418\n",
      "e value: 8.08932e-82\n",
      "MTAKQVAALAKAGEPIR-KSDGKGLY--FVVPDSGSPYWALRYSVDGKRKQMTLGQYADMSLADARTEAEVFKRE...\n",
      "+T  Q+ A  KAG     KSDG GLY  +   D+ SP W  RY   GK + MT+G Y  +SLA AR         ...\n",
      "LTDIQIKAWIKAGTRFEGKSDGNGLYLRYRTTDT-SPSWRFRYRFAGKPRVMTIGAYGVISLAKAREITRELAAR...\n",
      "kk419\n",
      "kk419\n",
      "kk419\n",
      "kk419\n",
      "kk398\n",
      "kk402\n",
      "kk402\n",
      "kk398\n",
      "kk401\n",
      "kk401\n",
      "kk401\n",
      "kk401\n",
      "kk412\n",
      "kk412\n",
      "kk412\n",
      "kk412\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk404\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk416\n",
      "kk418\n",
      "kk418\n",
      "kk418\n",
      "kk418\n",
      "kk416\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk418\n",
      "kk418\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk402\n",
      "kk402\n",
      "kk396\n",
      "kk396\n",
      "kk396\n",
      "kk396\n",
      "kk385\n",
      "kk396\n",
      "kk385\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk385\n",
      "kk395\n",
      "kk396\n",
      "kk396\n",
      "kk396\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk394\n",
      "kk394\n",
      "kk395\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk400\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk386\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk404\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk399\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk385\n",
      "kk404\n",
      "kk417\n",
      "kk396\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk411\n",
      "kk392\n",
      "kk404\n",
      "kk404\n",
      "kk394\n",
      "kk393\n",
      "kk393\n",
      "kk393\n",
      "kk401\n",
      "kk391\n",
      "kk393\n",
      "kk401\n",
      "kk333\n",
      "kk390\n",
      "kk387\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk394\n",
      "kk389\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk392\n",
      "kk401\n",
      "kk401\n",
      "kk401\n",
      "kk401\n",
      "kk390\n",
      "kk396\n",
      "kk393\n",
      "kk393\n",
      "kk393\n",
      "kk393\n",
      "kk397\n",
      "kk397\n",
      "kk391\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk396\n",
      "kk396\n",
      "kk396\n",
      "kk390\n",
      "kk389\n",
      "kk389\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk393\n",
      "kk393\n",
      "kk391\n",
      "kk391\n",
      "kk391\n",
      "kk391\n",
      "kk391\n",
      "kk389\n",
      "kk389\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk390\n",
      "kk401\n",
      "kk401\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk420\n",
      "kk396\n",
      "kk272\n",
      "kk407\n",
      "kk398\n",
      "kk369\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk427\n",
      "kk427\n",
      "kk427\n",
      "kk427\n",
      "kk390\n",
      "kk393\n",
      "kk396\n",
      "kk396\n",
      "kk396\n",
      "kk396\n",
      "kk396\n",
      "kk396\n",
      "kk414\n",
      "kk414\n",
      "kk414\n",
      "kk414\n",
      "kk414\n",
      "kk414\n",
      "kk414\n",
      "kk407\n",
      "kk407\n",
      "kk400\n",
      "kk398\n",
      "kk398\n",
      "kk400\n",
      "kk421\n",
      "kk421\n",
      "kk421\n",
      "kk421\n",
      "kk421\n",
      "kk421\n",
      "kk421\n",
      "kk399\n",
      "kk399\n",
      "kk399\n",
      "kk399\n",
      "kk399\n",
      "kk399\n",
      "kk399\n",
      "kk399\n",
      "kk399\n",
      "kk399\n",
      "kk391\n",
      "kk412\n",
      "kk412\n",
      "kk401\n",
      "kk394\n",
      "kk421\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk388\n",
      "kk417\n",
      "kk417\n",
      "kk417\n",
      "kk417\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk407\n",
      "kk402\n",
      "kk407\n",
      "kk403\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk420\n",
      "kk403\n",
      "kk403\n",
      "kk403\n",
      "kk403\n",
      "kk404\n",
      "kk407\n",
      "kk420\n",
      "kk404\n",
      "kk404\n",
      "Analyzing int_dusA\n",
      "* Query #1 is Int_dusA_VchOYP6G08_NZ_NMSY01000009*(length=366) \n",
      "kk362\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1057758 PGICJAHO_01842_chr_tig27_Res13-Lact-PEB03-33_chromosome\n",
      "length: 362\n",
      "e value: 7.8817e-152\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRTE--KPTIRELI...\n",
      "MA ++IEKR+ + G +R+RCTV VK+G   +YRE+RTF K++ AK WG   +++L+T+G+P   +  K  + +L+...\n",
      "MAYYSIEKRQRADGTLRYRCTVGVKEGGKYIYRENRTFSKQAHAKTWGAGRVAELETNGVPTTNDLNKMVVGDLL...\n",
      "kk362\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|923240 FJAJKMPC_00078_chr_tig3_Res13-Lact-LER1-21_chromosome\n",
      "length: 362\n",
      "e value: 7.8817e-152\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRTE--KPTIRELI...\n",
      "MA ++IEKR+ + G +R+RCTV VK+G   +YRE+RTF K++ AK WG   +++L+T+G+P   +  K  + +L+...\n",
      "MAYYSIEKRQRADGTLRYRCTVGVKEGGKYIYRENRTFSKQAHAKTWGAGRVAELETNGVPTTNDLNKMVVGDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|599795 MAOCHLDM_01632_chr_tig19_Res13-Fini-PEB21-32_chromosome\n",
      "length: 365\n",
      "e value: 6.23452e-143\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD HG+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLHGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|586408 EFLMONKE_02180_chr_tig42_Res13-Fini-PEB05-34_chromosome\n",
      "length: 365\n",
      "e value: 8.02624e-142\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|461152 LHCCHGGM_00875_chr_tig21_Res13-Croi-PEB07-04_chromosome\n",
      "length: 365\n",
      "e value: 8.02624e-142\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|457343 PAFFDKNG_01576_chr_tig33_Res13-Croi-PEB06-38_chromosome\n",
      "length: 365\n",
      "e value: 8.02624e-142\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|216356 NNAPDDJL_02835_chr_tig174_R13-CV1-pFin-02_chromosome\n",
      "length: 365\n",
      "e value: 1.18913e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|1305458 HDOFKCAA_00877_chr_tig18_Res13-Sevr-PEA12-35_chromosome\n",
      "length: 365\n",
      "e value: 1.54539e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|840498 LAGMDHBD_01668_chr_tig52_Res13-Lact-ER07-10_chromosome\n",
      "length: 365\n",
      "e value: 1.54539e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|834368 NKDPHGBB_00555_chr_tig14_Res13-Lact-ER07-09_chromosome\n",
      "length: 365\n",
      "e value: 1.54539e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|591061 FDKELFNH_02206_chr_tig44_Res13-Fini-PEB15-40_chromosome\n",
      "length: 365\n",
      "e value: 1.54539e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|232501 GCBIHENL_04060_chr_tig120_R13-CV1-pFin-05_chromosome\n",
      "length: 365\n",
      "e value: 2.58172e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|269230 ALHBKONJ_03888_chr_tig100_R13-CV2-pWea-03_chromosome\n",
      "length: 365\n",
      "e value: 2.72659e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|258978 NMLGNJJG_03631_chr_tig73_R13-CV2-pWea-01_chromosome\n",
      "length: 365\n",
      "e value: 2.72659e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|100617 KFEJGMCL_01947_chr_tig42_R13-AF22-pFin-03_chromosome\n",
      "length: 365\n",
      "e value: 3.07454e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|91735 JNMPAFIB_01865_chr_tig54_R13-AF22-pFin-01_chromosome\n",
      "length: 365\n",
      "e value: 3.07454e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|622188 CAMEKIGI_00441_chr_tig26_Res13-Fini-PEC06-39_chromosome\n",
      "length: 365\n",
      "e value: 4.03949e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|255203 FHBEHDNA_04259_chr_tig180_R13-CV1-pSow-05_chromosome\n",
      "length: 365\n",
      "e value: 4.03949e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|241382 NGKPAMAB_03907_chr_tig174_R13-CV1-pSow-02_chromosome\n",
      "length: 365\n",
      "e value: 4.03949e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|236856 LEAINBDH_03815_chr_tig216_R13-CV1-pSow-01_chromosome\n",
      "length: 365\n",
      "e value: 4.03949e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk365\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|108868 LFOMNGJH_01408_chr_tig39_R13-AF22-pFin-05_chromosome\n",
      "length: 365\n",
      "e value: 4.03949e-141\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRT--EKPTIRELI...\n",
      "MA + IEKR  S G  R+RC V +K+   + YRES+TF K + AK WGT  + +LD +G+P     +  T+R+L+...\n",
      "MAYYNIEKRLKSDGTPRYRCNVIIKEKGVITYRESKTFPKHAHAKTWGTQKVMELDLYGIPSSNAVDGLTVRDLL...\n",
      "kk366\n",
      "****Alignment****\n",
      "sequence: gnl|BL_ORD_ID|349924 CGFKMMEF_01791_chr_tig16_Res13-Abat-PEB01-P1-04-A_chromosome\n",
      "length: 366\n",
      "e value: 3.64394e-126\n",
      "MASFTIEKRKTSKGIIRFRCTVRVKKGNDVVYRESRTFGKRSDAKEWGTLLMSKLDTHGLPGRTEKPTIRELIL-...\n",
      "MA FTIEKR  S G  R+RCTV VKK    VYRE++TF K + AK WG   ++ L+ HG+  + ++    ELIL ...\n",
      "MAFFTIEKRLRSDGTARYRCTVGVKKNGKYVYRENQTFSKNTLAKSWGAKRVAYLEEHGIQHKAQEGA-PELILT...\n",
      "kk468\n",
      "kk109\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk382\n",
      "kk315\n",
      "kk315\n",
      "kk315\n",
      "kk315\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk389\n",
      "kk389\n",
      "kk389\n",
      "kk384\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk374\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk384\n",
      "kk357\n",
      "kk357\n",
      "kk357\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n",
      "kk398\n"
     ]
    }
   ],
   "source": [
    "# This code provide nice output of very high-quality aligments\n",
    "# And give the names of the strains that contains the query sequence\n",
    "# Will work only if the query contains multiples versions of the same gene, e.g. many slightly related intimins\n",
    "# Fragile code : The code search for a pattern in alignment titles\n",
    "# Because our strains begins by Res13 or R13\n",
    "# Adjust that pattern to reflect your data\n",
    "\n",
    "\n",
    "# First grab the bastp output files in the xml folder\n",
    "xml_files = get_files (\"./xml\", \"xml\")\n",
    "\n",
    "p = re.compile('.+\\_(R[es]*13[^_]+)\\_(\\S+)')  # a pattern for identifying our strains in alignemnt titles\n",
    "E_VALUE_THRESH = 1e-80\n",
    "S = set()\n",
    "for f in xml_files:\n",
    "    virulence_gene_name = os.path.basename(f).replace('.xml','')\n",
    "    print (\"Analyzing\", virulence_gene_name, sep=' ')\n",
    "    result_handle = open(f)\n",
    "    from Bio.Blast import NCBIXML\n",
    "    blast_records = NCBIXML.parse(result_handle)\n",
    "    for blast_record_i, blast_record in enumerate(blast_records, start=1):\n",
    "        print('* Query #' + repr(blast_record_i) + ' is ' + blast_record.query + '*', end='')\n",
    "        print('(length=' + repr(blast_record.query_letters) + ') ', end='\\n')\n",
    "        \n",
    "        for alignment in blast_record.alignments:            \n",
    "            for hsp in alignment.hsps:\n",
    "                # restrict to near full_length alignment\n",
    "                if ((hsp.expect < E_VALUE_THRESH) & (hsp.align_length/blast_record.query_letters > 0.95)):\n",
    "                    #print(\"Hit found in\", f)\n",
    "                    print(\"****Alignment****\")\n",
    "                    print(\"sequence:\", alignment.title)\n",
    "                    m = p.match(alignment.title) # look if the pattern can be found in the alignment title\n",
    "                    if m:\n",
    "                        S.add( (virulence_gene_name, m.group(1), m.group(2)) ) # add a tuple to our set S\n",
    "                    print(\"length:\", alignment.length)\n",
    "                    print(\"e value:\", hsp.expect)\n",
    "                    print(hsp.query[0:75] + \"...\")\n",
    "                    print(hsp.match[0:75] + \"...\")\n",
    "                    print(hsp.sbjct[0:75] + \"...\")\n",
    "                    \n",
    "                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('int_yicC', 'R13-AF11-pFin-04', 'chromosome'),\n",
       " ('int_yicC', 'R13-AF12-pSow-01', 'chromosome'),\n",
       " ('int_yicC', 'R13-AF21-pWea-04', 'chromosome'),\n",
       " ('int_dusA', 'R13-AF22-pFin-01', 'chromosome'),\n",
       " ('int_dusA', 'R13-AF22-pFin-03', 'chromosome'),\n",
       " ('int_dusA', 'R13-AF22-pFin-05', 'chromosome'),\n",
       " ('int_yicC', 'R13-AF22-pSow-03', 'chromosome'),\n",
       " ('int_yicC', 'R13-AF22-pSow-04', 'chromosome'),\n",
       " ('int_yicC', 'R13-AF22-pSow-05', 'chromosome'),\n",
       " ('int_yicC', 'R13-AF31-pSow-02', 'chromosome'),\n",
       " ('int_yicC', 'R13-AF31-pSow-05', 'chromosome'),\n",
       " ('int_dusA', 'R13-CV1-pFin-02', 'chromosome'),\n",
       " ('int_dusA', 'R13-CV1-pFin-05', 'chromosome'),\n",
       " ('int_dusA', 'R13-CV1-pSow-01', 'chromosome'),\n",
       " ('int_dusA', 'R13-CV1-pSow-02', 'chromosome'),\n",
       " ('int_yicC', 'R13-CV1-pSow-02', 'chromosome'),\n",
       " ('int_yicC', 'R13-CV1-pSow-04', 'chromosome'),\n",
       " ('int_dusA', 'R13-CV1-pSow-05', 'chromosome'),\n",
       " ('int_dusA', 'R13-CV2-pWea-01', 'chromosome'),\n",
       " ('int_dusA', 'R13-CV2-pWea-03', 'chromosome'),\n",
       " ('int_yicC', 'R13-CV2-pWea-05', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Abat-PEB01-P1-04-A', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Croi-PEB06-38', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Croi-PEB07-04', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Croi-PEB17-01', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Fini-PEB05-34', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Fini-PEB15-40', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Fini-PEB21-32', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Fini-PEC06-39', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Fini-PEC06-39', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Lact-ER07-09', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Lact-ER07-10', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-LEB3-02', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-LEB3-12', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-LEB3-20', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-LEB3-27', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Lact-LER1-21', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-LER2-35-b', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEA02-11', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEA13-09', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEA13-10', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEA13-11', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEA13-17', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEA13-25', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEA13-26', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Lact-PEB03-33', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEB07-13-B', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEB17-18', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEB18-16', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Lact-PEC11-32', 'chromosome'),\n",
       " ('int_dusA', 'Res13-Sevr-PEA12-35', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Sevr-PEB15-13', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Sevr-PEC09-36', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Sevr-PEC09-46', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Sevr-PEC09-47', 'chromosome'),\n",
       " ('int_yicC', 'Res13-Sevr-PER08-38', 'chromosome')]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Now discover which strains have the complete query genes\n",
    "L = list(S) # Change our Set to a mutable List\n",
    "L.sort(key=lambda tup: tup[1]) # Sort by strain\n",
    "L"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/brouardjs/data/old_fujitsu_laptop/GTlab/MS_minION/virulence_gene_detection'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query: Int_dusA_VchOYP6G08_NZ_NMSY01000009\n",
      "  Hit: PGICJAHO_01842_chr_tig27_Res13-Lact-PEB03-33_chromosome (362)\n",
      " HSPs: ----  --------  ---------  ------  ---------------  ---------------------\n",
      "          #   E-value  Bit score    Span      Query range              Hit range\n",
      "       ----  --------  ---------  ------  ---------------  ---------------------\n",
      "          0  7.9e-152     436.03     361          [0:359]                [0:361]\n"
     ]
    }
   ],
   "source": [
    "from Bio import SearchIO\n",
    "blast_qresult = SearchIO.read(\"xml/int_dusA.xml\", \"blast-xml\")\n",
    "blast_hit = blast_qresult[0]                                                  \n",
    "print(blast_hit)    \n",
    "                                                                       \n",
    "                                                                       \n",
    "                                                                       \n",
    "                                                                       "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### End of analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Extra pieces of code"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Code below explain how the bfp query was created"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''Now we are looking for bundle forming pili genes (bfp) in our strains\n",
    "We will use the Genbank Accession number AB024946 (S.abraham et al. )\n",
    "Go get the bfp gene in this Genbank record of this plasmid\n",
    "and then prepare a multifasta file with bfp genes present on a plasmid\n",
    "'''\n",
    "from Bio import Entrez\n",
    "Entrez.email =\"jean-simon.brouard@canada.ca\"\n",
    "handle = Entrez.efetch(db=\"nucleotide\", id='AB024946', rettype=\"gb\", retmode=\"text\")\n",
    "with open('./AB024946.gb', \"w\") as fp:\n",
    "    fp.write(handle.read())\n",
    "    \n",
    "# Read only 1 record\n",
    "record = SeqIO.read(\"AB024946.gb\", \"genbank\")\n",
    "\n",
    "# Prepare a fasta file for writing\n",
    "with open('queries/bfp.fasta', \"w\") as output_handle:\n",
    "        \n",
    "    for i_feature, feature in enumerate(record.features):\n",
    "        if feature.type == 'CDS':\n",
    "            if 'gene' in feature.qualifiers: # if a gene tag is present use it the name\n",
    "                if 'bfp' in feature.qualifiers['gene'][0]:\n",
    "                    feature_id = feature.qualifiers['gene'][0]\n",
    "                    # Create the sequence object\n",
    "                    my_seq = Seq(feature.qualifiers['translation'][0],IUPAC.protein)\n",
    "                \n",
    "                    # Create a SeqRecord object with the protein sequence\n",
    "                    simple_seq_r = SeqRecord(my_seq, id=feature_id, description = '')\n",
    "                    SeqIO.write(simple_seq_r, output_handle, \"fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Just to show how we can iterate on blast record descriptions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Code for outputing the description lines of blast output\n",
    "# We have aligments, bit scores and E-value\n",
    "# Note that there is 1 BLAST record object/query if many sequences are given as query\n",
    "# In addition, hits with good E-value are found in all strains, but one will want to look\n",
    "# at the aligments to determine if the eae gene is present or not (see the cell below)!\n",
    "E_VALUE_THRESH = 1e-80\n",
    "for f in xml_files:\n",
    "    print (\"Analyzing\", f, sep=' ')\n",
    "    result_handle = open(f)\n",
    "    from Bio.Blast import NCBIXML\n",
    "    blast_records = NCBIXML.parse(result_handle)\n",
    "    for blast_record in blast_records:\n",
    "        for description in blast_record.descriptions:\n",
    "            print(description)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
